### PR TITLE
Fix sparse indexes

### PR DESCRIPTION
**Pull request by guilhermearaujo**

### CODE REVIEW

File: drf_kit/models.py [hunk 1]
```diff
@@ -128,24 +127,33 @@ class OrderedModelMixin(_OrderedModelBase):
...
-        clash = self.get_ordering_queryset().filter(**{self.order_field_name: new_order}).exclude(pk=self.pk).first()
-        if not clash:
-            return
+        order = getattr(self, self.order_field_name)
 
-        setattr(clash, self.order_field_name, new_order + 1)
-        clash.save()
+        group = list(self.get_ordering_queryset().exclude(id=self.pk))
+
+        if order:  # when updating
```
Comment (joaodaher) [inline drf_kit/models.py:144]: Should be `if order is not None` to handle cases that `order=0` intentionally.

File: drf_kit/models.py [hunk 2]
```diff
@@ -128,24 +127,33 @@ class OrderedModelMixin(_OrderedModelBase):
...
+            group.insert(max(0, self.order), self)
```
Comment (joaodaher) [inline drf_kit/models.py:145]: Should be `group.insert(max(0, order), self)` to handle custom order field name

**Review by joaodaher — CHANGES_REQUESTED** [review]

File: drf_kit/models.py (hunk 1 above)
Comment (joaodaher) [reply drf_kit/models.py:144]: It'd be good to have an additional test for this too.
"Move an item from 3 to 0"

Comment (guilhermearaujo) [reply drf_kit/models.py:144]: Good catch!

**Review by joaodaher — APPROVED** [review]

**Pull request merged**